hibernate-core/src/main/java/org/hibernate/query/spi/AbstractSelectionQuery.java

```diff
@@ -378,6 +381,22 @@ public List<R> list() {
 		}
 	}
 
+	@Override
+	public List<R> uniqueList() {
+		final List<R> list = list();
+		if ( list.isEmpty() ) {
+			return list;
+		}
+		Set<R> dict = CollectionHelper.linkedSetOfSize( list.size() );
+		List<R> uniqueList = new ArrayList<>( list.size() );
+		for ( R listElement : list ) {
+			if ( dict.add( listElement ) ) {
+				uniqueList.add( listElement );
+			}
+		}
+		return uniqueList;
```

Comment:
I was thinking about using a new (to be added) method `AbstractSelectionQuery#doList(ListResultsConsumer.UniqueSemantic)` that delegates to `SelectQueryPlan#performList(DomainQueryExecutionContext, ListResultsConsumer.UniqueSemantic)` (additional method or alter the existing one since the contract is incubating) so we can use our existing `ListResultsConsumer` code for handling this. Wdyt @sebersole and @dreab8?

Reply:
I think I misunderstood the requirements. The gist of the ticket is to use the `select distinct xxx' from db server side rather than such superficial post processing. For that reason, I closed this PR. Thanks for the feedback.